index.bs

```diff
@@ -12,23 +12,55 @@ Abstract: This API defines an API surface for controlling how audio is rendered
+Additionally, some platforms automatically manage a site's audio session
+based on media playback and the APIs used to play audio.
+However, this behavior might not always align with user expectations.
+This API allows developers to override the default behavior and gain more control over an audio session.
 
-# The {{AudioSession}} interface # {#audiosession-interface}
+# Concepts # {#concepts}
+
+An <dfn>audio session</dfn> represents the playback of audible media.
+An [=audio session=] can be of a particular [=audio session/type=], and be in a particular [=audio session/state=].
+An [=audio session=] manages the audio for a set of sources and sinks, named audio session <dfn data-lt="element" for="audio session">elements</dfn>.
+
+An audio session [=audio session/element=] has a number of properties:
+
+* A <dfn data-lt="default type" for="audio session">default type</dfn>, which is used to compute the [=audio session=] [=audio session/type=], in case of "{{AudioSessionType/auto}}".
+* An <dfn>audible flag</dfn>, which is either `true` if the element is playing/recording audio or `false` otherwise.
+
+An audio session [=audio session/element=] is an <dfn>audible element</dfn> if its [=audible flag=] is `true`.
 
-An <dfn>audio session</dfn> represents the playback of auditory media.
-An audio session can be of particular [=audio session/type=] and in a given [=audio session/state=].
+Additionaly, an audio session [=audio session/element=] has associated steps for dealing with various state changes. By default, each of these steps is empty list of steps:
 
-The {{AudioSession}} is the main interface for this API, which is accessed through the {{Navigator}} interface (see [[#extensions-to-navigator]]).
+* <dfn data-lt="element update steps" for="audio session">Element update steps</dfn>, which are run whenever the audio session state changes.
+* <dfn data-lt="element suspend steps" for="audio session">Element suspend steps</dfn>, which are run when the audio session state moves from {{AudioSessionState/active}} to either {{AudioSessionState/interrupted}} or {{AudioSessionState/inactive}}.
+* <dfn data-lt="element resume steps" for="audio session">Element resume steps</dfn>, which are run when audio session state moves from {{AudioSessionState/interrupted}} to {{AudioSessionState/active}}.
+
+A [=top-level browsing context=] has a <dfn>selected audio session</dfn>. The [=selected audio session=] may be [=update the selected audio session|updated=] dynamically.
+A [=top-level browsing context=] is said to have <dfn lt="audio focus" for="audio session">audio focus</dfn> if its [=selected audio session=] is not `null` and its state is {{AudioSessionState/active}}.
+
+<div class=note>
+User agents can decide whether to allow several [=top-level browsing context=] to have [=audio focus=], or to enforce that only a single [=top-level browsing context=] has [=audio focus=] at any given time.</div>
```

Comment:
Could you provide an example of a situation where multiple audio focus is necessary? Is it premature to mention audio focus at this stage? It might be more effective to include a full paragraph defining what audio focus is. Perhaps we should explain it thoroughly after introducing the different types?

For instance, does an audio session with the ambient type qualify as having audio focus, given that it can play and mix with other types? 

Additionally, should we consider adding an attribute to the AudioSession to indicate whether it is a selected audio session?

Reply:
On desktop, UAs tend to allow multiple tabs to render audio, similar to several native applications playing audio at the same time (without being ambient).

> For instance, does an audio session with the ambient type qualify as having audio focus, given that it can play and mix with other types?

It does not qualify.


> should we consider adding an attribute to the AudioSession to indicate whether it is a selected audio session?

We could consider it but I am a bit reluctant to do this for now until we have a good use case.

Reply:
Thanks for clarification, I think it would be worth to provide some examples here. But we could do that in following up PRs.

Reply:
I filed https://github.com/w3c/audio-session/issues/29.